### DISCUSSION TITLE
App-only (zero touch) deployment capability - available now

### DISCUSSION THREAD

**Discussion by dkershaw10** [microsoftgraph/msgraph-bicep-types#106] [Announcements]
Hi folks,

It's been a while coming, but it's finally here.
Please try out being able to push a deployment of a Bicep file (with Graph/Entra) resources using a service principal.

Basic instructions are available here in this [app-only quickstart](https://github.com/microsoftgraph/msgraph-bicep-types/blob/main/docs/msgraph-bicep-docs/conceptual/quickstart-create-bicep-zero-touch-mode.md).  Please note that this quick start deploys the Bicep file created in a previous [getting started quickstart](https://github.com/microsoftgraph/msgraph-bicep-types/blob/main/docs/msgraph-bicep-docs/conceptual/quickstart-create-bicep-interactive-mode.md)

Please let us know ASAP if you run into any problems or want to give us any feedback.

While it's not there yet, we plan to add another How-to on app-only deployments, including a bit on privileges and permissions and code samples to acquire tokens to call the ARM deployment service to deploy Bicep files.

Thanks!

**Comment by rajyraman** [comment]
@dkershaw10 - ~Both links don't work.~ Found this link https://github.com/microsoftgraph/msgraph-bicep-types/blob/dkershaw10-publicdocspreview/docs/msgraph-bicep-docs/conceptual/quickstart-create-bicep-zero-touch-mode.md. All OK.

**Comment by dkershaw10** [comment]
@rajyraman - Sorry should have updated and will add an announcement and close this discussion. The public preview is out, and the docs are public.  The doc links above were only hosted here temporarily to get early feedback.  They've since been removed in favour of the public docs.

See https://aka.ms/graphbicep
Also the repo's home page Readme should point to the public docs.
